src/templates/hl7v2/ADT_A01.hbs

```diff
@@ -3,50 +3,83 @@
     "type": "transaction",
     "entry": [
         {{#with (getFirstSegments msg.v2 'PID' 'PD1' 'PV1' 'PV2' 'PR1' 'AVR')}}
-    
-            {{>Resources/Patient.hbs PID=PID PD1=PD1 NK1=NK1 ID=(generateUUID PID-3-1)}},
-            {{>Resources/Encounter.hbs PV1=PV1 PV2=PV2 ID=(generateUUID PV1)}},
-            {{>Resources/Procedure.hbs PR1=PR1 ID=(generateUUID PR1)}},
-            {{>References/Encounter/subject.hbs ID=(generateUUID PV1) REF=(concat 'Patient/' (generateUUID PID-3-1))}}, 
-            {{>References/Procedure/subject.hbs ID=(generateUUID PR1) REF=(concat 'Patient/' (generateUUID PID-3-1))}},	
-            
+
+            {{#if PID-3-1}},
+                {{>Resources/Patient.hbs PID=PID PD1=PD1 NK1=NK1 ID=(generateUUID PID-3-1)}},
+            {{/if}},
+
+            {{#if PV1}},
```

Comment:
It seems PV1, PR1 comes from "getFirstSegments". Could we update "getFirstSegments" as well to avoid getting null or undefined?

Reply:
Might a little different, 'cause this statement actually will not care about the return value from "getFirstSegments", it just declare a usage.

Even the "getFirstSegments" does not return the null or undefined value, the subsequent templates will also say that "I will use this segment, and I don't care whether it's undefined or not".